### PR TITLE
Donkpockets 4: Return of the Mousepockets

### PR DESCRIPTION
**Pull request by Coconutwarrior97**
## About The Pull Request
image

Adds mouse pockets and a crab pocket to the game!

## Why It's Good For The Game

Mouse pockets are made from tasty easily available maint mice! Also theres a crab pocket which is cool.

## Changelog
:cl:
add: Adds mouse pockets + crab pocket for crafting.
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm [hunk 1]
```diff
@@ -399,6 +399,67 @@
...
+		/obj/item/food/pastrybase = 1,
+		/obj/item/food/deadmouse = 1
+	)
+	result = /obj/item/food/donkpocket/donkmousebrown
+	subcategory = CAT_PASTRY
+
+/datum/crafting_recipe/food/donkpocket/donkmousebrown/check_requirements(mob/user, list/collected_requirements) //Important override so that a specific icon_state is required for the specific mouse pocket to be crafted. (So you cant make brown mouse pockets with white mice).
+	var/obj/item/food/deadmouse = collected_requirements[/obj/item/food/deadmouse][1]
+	var/mouse_state = deadmouse.icon_state //Getting the icon_state of the deadmouse and setting it equal to mouse_state.
+	if(mouse_state == "mouse_brown_dead")
```
Comment (Jared-Fogle) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm:415]: When is this not true? Admin intervention?

I think either way, I'd rather this check `initial(deadmouse.icon_state)` (after correctly checking `var/obj/item/food/deadmouse` to `var/obj/item/food/deadmouse/deadmouse`).

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm [hunk 2]
```diff
@@ -399,6 +399,67 @@
 	result = /obj/item/food/donkpocket/gondola
 	subcategory = CAT_PASTRY
 
+/datum/crafting_recipe/food/donkpocket/donkmousebrown
```
Comment (Jared-Fogle) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm:402]: I don't really like having separate recipe per mice, our crafting menu is already not very great as is.

This should instead be a general recipe for any `/obj/item/food/deadmouse`, which creates a generic `/obj/item/food/donkpocket/deadmouse`. Then, *that* re-implements `/CheckParts`. That way, you can dynamically update the sprite depending on the mouse.

Search CheckParts across the codebase to find existing crafting recipes that do this.

Comment (Jared-Fogle) [reply code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm:402]: If you want to keep the different tastes and descriptions and whatnot, you still can with this, you should be able to replace the item in CheckParts.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I'd like the recipes to be combined into one, creating an abstract dead mouse donk pocket that decides then and there.

**Comment by Farquaar** [comment]
Wow those sprites look fantastic

**Comment by maxymax13** [comment]
delicious

**Review by Twaticus — APPROVED** [review]

**Comment by Escamado** [comment]
they should make lizards happy, since all donkpockets make them disgust since its a pastry.

**Comment by optimumtact** [comment]
@Coconutwarrior97 stop playing vrchat and finish this PR instead

**Comment by Mothblocks** [comment]
this has been untouched for a month silly

**Pull request closed without merging**